recipes/transformer-engine/meta.yaml

```diff
@@ -0,0 +1,104 @@
+{% set version = "1.13" %}
+{% set torch_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
+
+{% if cuda_compiler_version in (None, "None", True, False) %}
+{% set cuda_major = 0 %}
+{% else %}
+{% set cuda_major = environ.get("cuda_compiler_version", "11.8").split(".")[0] | int %}
+{% endif %}
+
+package:
+  name: transformer-engine
+  version: {{ version }}
+
+source:
+  - url: https://github.com/NVIDIA/TransformerEngine/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 90dd8f694cba29707c502a4f4d12871cb9d4c1f157d329f22cb3d382de8468ae
+    patches:
+      - 001-cmake.patch
+
+  - url: https://github.com/NVIDIA/cudnn-frontend/archive/refs/tags/v1.8.0.tar.gz
+    sha256: 5907bf8baa8f3b74ad430d54027aadd14a0a61166ae4b00c2daece0df384409e
+    folder: 3rdparty/cudnn-frontend
+
+  - url: https://github.com/google/googletest/archive/refs/tags/v1.14.0.tar.gz
+    sha256: 8ad598c73ad796e0d8280b082cebd82a630d73e73cd3c70057938a6501bba5d7
+    folder: 3rdparty/googletest
+
+build:
+  number: 0
+
+outputs:
+  - name: transformer-engine-torch
+    version: {{ version }}
+    script: build_te.sh
+
+    build:
+      skip: true  # [not (cuda_compiler_version or "").startswith("12")]
+      skip: true  # [not (linux64 or aarch64)]
+      script_env:
+        - CUDA_HOME={{ BUILD_PREFIX }}
+        - NVTE_FRAMEWORK=pytorch
+      run_exports:
+        - {{ pin_subpackage('transformer-engine-torch', max_pin='x.x') }}
+
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - pytorch                                # [build_platform != target_platform]
+        - pytorch =*={{ torch_proc_type }}*      # [build_platform != target_platform]
+        - {{ stdlib('c') }}
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}  # [cuda_compiler_version not in (undefined, 'None')]
+        - cuda-nvvm
+        - cmake >=3.21
+        - ninja
+
+      host:
+        - python
+        - pip
+        - pybind11
+        - pytorch
+        - pytorch =*={{ torch_proc_type }}*
+        - setuptools
+        - pydantic
+        - importlib-metadata >=1.0
+        - packaging
+        - cuda-driver-dev
+        - cuda-cudart-dev
+        - cuda-nvrtc-dev
+        - cuda-nvtx-dev
+        - cuda-nvml-dev
+        - libcublas-dev
+        - libcusparse-dev
+        - libcusolver-dev
```

Comment:
There are various unused dependencies at runtime from these CUDA libraries. Are we using dlopen or statically linking? Or are the deps actually unused?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1235681&view=logs&j=df9408f7-e795-500f-143d-aa7735a0ed8b&t=47addc80-ccbf-566b-ee3a-66cdb8436937&l=2494

Reply:
Yeah good question; they might only be needed at build time. Would I move these from `host` to `build` then? I'm pretty sure pydantic just wasn't exercised during the import tests, added an additional import to hopefully hit that.

Reply:
As discussed offline, we are assuming these packages are dlopen'd and will decide whether to make them optional at a later time.